src/java/org/apache/cassandra/db/filter/RowFilter.java

```diff
@@ -261,6 +263,11 @@ public RowFilter withoutExpressions()
         return withNewExpressions(Collections.emptyList());
     }
 
+    public RowFilter restrict(Predicate<Expression> filter)
```

Comment:
nit/naming: Perhaps this could just be another `withNewExpressions(Predicate<Expression> predicate)` overload? (Would read like `rowFilter.withNewExpressions(predicate)`?)

Reply:
I would be more in favour of changing both this method and the one above to `without`. That would make them all consistent because they are all then returning a `RowFilter` without the expression / predicate parameter and in the case of no parameter an empty `RowFilter`. 

I am also not particularly happy with the current `withoutExpressions` method returning a `Collections.emptyList`. This, to my mind, breaks the contract of the `RowFilter`. I can accept that the `NONE` or `none()` can return a static immutable `RowFilter` but `withoutExpressions` is modifying an existing `RowFilter` therefore should allow following changes to it.

WDYT?

Reply:
No objection from me, as long as things are consistent, and it's clear whether the filter is filtering things out (subtract) or keeping things (retain).

Reply:
I'd also be fine changing `restrict()` -> `withMatching()` or something like that, given it retains the expressions that match the `Predicate`, right?